app/services/invoices/create_service.rb

```diff
@@ -2,36 +2,41 @@
 
 module Invoices
   class CreateService < BaseService
-    def initialize(subscription:, timestamp:)
-      @subscription = subscription
+    def initialize(subscriptions:, timestamp:)
+      @subscriptions = subscriptions
       @timestamp = timestamp
+      @customer = subscriptions&.first&.customer
+      @currency = subscriptions&.first&.plan&.amount_currency
 
       super(nil)
     end
 
     def create
       ActiveRecord::Base.transaction do
-        invoice = Invoice.find_or_create_by!(
-          subscription: subscription,
-          from_date: from_date,
-          to_date: to_date,
-          charges_from_date: charges_from_date,
-          issuing_date: issuing_date,
+        invoice = Invoice.create!(
```

Comment:
Do you think that is fine to use only `create!` here? SInce we removed some fields I was afraid that we would return wrong invoice in some cases. (e.g. one subscription was terminated and billed for pay in arrear in the same day when invoice for another subscription is billed)